services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/NodeManagerImpl.java

```diff
@@ -639,6 +639,14 @@ public EntityHeader getNodeHeader(UserInfo userInfo, String entityId)
 		authorizationManager.hasAccess(userInfo, entityId, ACCESS_TYPE.READ).checkAuthorizationOrElseThrow();
 		return nodeDao.getEntityHeader(entityId);
 	}
+
+	@Override
+	public EntityHeader getNodeHeader(UserInfo userInfo, String entityId, Long versionNumber)
+			throws NotFoundException, DatastoreException, UnauthorizedException {
+		UserInfo.validateUserInfo(userInfo);
+		authorizationManager.hasAccess(userInfo, entityId, ACCESS_TYPE.READ).checkAuthorizationOrElseThrow();
```

Comment:
Can we add a unit test for this?

Reply:
Done. Added unit tests for new method `getNodeHeader(UserInfo, String, Long)` and backfilled unit tests for existing method `getNodeHeader(UserInfo, String)`